### DISCUSSION TITLE
Which social network should I add next?

### DISCUSSION THREAD

**Discussion by sb0y** [sb0y/warp_beacon#2] [Polls]
Hey!
Which social network would you like to see added to the bot's functionality next?

**Comment by becauseimnoob** [comment]
1. Facebook,X (Twitter), TikTok,pinterest,spotify,soundcloud,
2. There is a YouTube video download feature in the existing code, this should be converted to mp3.
3.  Also, a progress bar is a must because you don't want to look at a loading.gif that keeps looping for hours. This is a bad experience for the user.  #34 
4. Also, a solution should be found for Instagram, accounts are getting banned very quickly.  @sb0y

**Reply by sb0y** [reply]
@becauseimnoob hi, feature #34 is done.
I’ve also done significant work on countering Instagram’s anti-bot protection mechanisms.
My current setup includes 4 IPv6/IPv4 proxies and two personal Instagram accounts — one of which is a sub-account registered with the same phone number.

**Reply by sb0y** [reply]
Also, just a quick note: YouTube no longer stores audio tracks in MP3 format — only in M4A.
So converting to MP3 would mean unnecessary re-encoding, lower audio quality, and slower processing. I believe it's better to keep the original M4A format.

**Reply by sb0y** [reply]
Also, don't forget to generate and persist a constant `device_id` to avoid triggering Instagram’s anti-bot protection.
You can use a script like this:

```python
import uuid
import secrets
import hashlib

def generate_uuid() -> str:
    return str(uuid.uuid4())

def generate_device_id(seed: str = None) -> str:
    raw = secrets.token_hex(8) if not seed else seed.encode()
    hex_part = hashlib.md5(raw).hexdigest()[:16]
    return f"android-{hex_part}"

print(generate_device_id("your_ig_login"))
```

Also, make sure you populate the `instagram.auth_details.device.uuids.device_id` field in your `accounts.json`.
Otherwise, the bot will generate a new one on every launch, which can trigger Instagram’s anti-bot mechanisms. Changing `device_id` too often increases the risk of getting flagged.

**Reply by sb0y** [reply]
Starting from version [2.6.89](https://github.com/sb0y/warp_beacon/releases/tag/2.6.89), you can also use the CLI command `warp_beacon --generate-device-id your_ig_login`. 
If you're running inside Docker:

```bash
cd <directory_with_docker_compose>
docker exec -it warp_beacon bash -c 'warp_beacon --generate-device-id your_ig_login'
```
This will generate a proper `device_id` automatically, linked to your IG login.
